src/background.js

```diff
@@ -0,0 +1,111 @@
+"use strict";
+
+/* global browser */
+const HAS_SEEN_VERSION = 1;
+const STUDY_URL = browser.extension.getURL("study.html");
+const SETTING_NAME = "trr";
+
+const stateManager = {
+  async getState() {
+    return await browser.experiments.settings.get(SETTING_NAME) || null;
+  },
+
+  async setState(stateKey) {
+    browser.experiments.settings.set(SETTING_NAME, stateKey);
+  },
+
+  getUserPrefKey(key) {
+    return `userPref_${key}`;
+  },
+
+  async setSetting() {
+    return browser.experiments.settings.add(SETTING_NAME);
+  },
+};
+
+const rollout = {
+  async init() {
+    browser.runtime.onMessage.addListener((...args) => this.handleMessage(...args));
+    await stateManager.setSetting();
+    const stateName = await stateManager.getState();
+    switch (stateName) {
+      case null:
+        if (await browser.experiments.settings.hasUnmodifiedPrerequisites(SETTING_NAME)) {
+          await stateManager.setState("loaded");
+          await this.show();
+        } else {
+          // If the user hasn't met the criteria clean up
+          browser.management.uninstallSelf();
```

Comment:
This will reset the user prefs through calling the UNINSTALL handler in api.js. Please make sure that doesn't happen, e.g. through defining a new state for this.

Reply:
:heavy_check_mark: Done, I moved this to use the new null clear state that is in the experiment.